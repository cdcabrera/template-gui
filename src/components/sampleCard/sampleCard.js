import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardHead, CardBody } from '@patternfly/react-core';
import { connect, reduxActions } from '../../redux';
import { helpers } from '../../common/helpers';

class SampleCard extends React.Component {
  componentDidMount() {
    const { getSampleData } = this.props;

    getSampleData();
  }

  render() {
    const { error, fulfilled, data, pending } = this.props;

    return (
      <Card>
        <CardHead>
          <h5>Sample</h5>
        </CardHead>
        <CardBody>
          Data ipsum... {(error && 'error') || (fulfilled && 'loaded') || (pending && 'pending')}{' '}
          {data.usage.length && JSON.stringify(data)}
        </CardBody>
      </Card>
    );
  }
}

SampleCard.propTypes = {
  error: PropTypes.bool,
  fulfilled: PropTypes.bool,
  getSampleData: PropTypes.func,
  data: PropTypes.shape({
    usage: PropTypes.array
  }),
  pending: PropTypes.bool
};

SampleCard.defaultProps = {
  error: false,
  fulfilled: false,
  getSampleData: helpers.noop,
  data: {
    usage: []
  },
  pending: false
};

const mapStateToProps = state => ({
  ...state.sample
});

const mapDispatchToProps = dispatch => ({
  getSampleData: () => dispatch(reduxActions.sample.getSampleData())
});

const ConnectedSampleCard = connect(
  mapStateToProps,
  mapDispatchToProps
)(SampleCard);

export { ConnectedSampleCard as default, ConnectedSampleCard, SampleCard };
