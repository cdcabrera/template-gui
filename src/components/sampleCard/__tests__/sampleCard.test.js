import React from 'react';
import { mount } from 'enzyme';
import { SampleCard } from '../sampleCard';

describe('SampleCard Component', () => {
  it('should render a non-connected component', () => {
    const props = {};

    const component = mount(<SampleCard {...props} />);

    expect(component).toMatchSnapshot('non-connected');
  });
});
