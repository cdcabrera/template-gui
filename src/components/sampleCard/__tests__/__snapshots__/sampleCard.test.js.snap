// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleCard Component should render a non-connected component: non-connected 1`] = `
<SampleCard
  data={
    Object {
      "usage": Array [],
    }
  }
  error={false}
  fulfilled={false}
  getSampleData={[Function]}
  pending={false}
>
  <Card>
    <article
      className="pf-c-card"
    >
      <CardHead>
        <div
          className="pf-c-card__head"
        >
          <h5>
            Sample
          </h5>
        </div>
      </CardHead>
      <CardBody>
        <div
          className="pf-c-card__body"
        >
          Data ipsum... 
           
          0
        </div>
      </CardBody>
    </article>
  </Card>
</SampleCard>
`;
